src/shared/schemas.ts

```diff
@@ -344,7 +348,14 @@ export class YamlSchemaHandler implements SchemaHandler {
         }
 
         if (mapping.schema) {
-            this.yamlExtension.assignSchema(mapping.uri, mapping.registry, resolveSchema(mapping.schema, schemas))
+            const schema = resolveSchema(mapping.schema, schemas)
+            if (!schema) {
+                getLogger().debug(
+                    `Could not assign schema ${mapping.schema} to ${mapping.uri}. Unable to find schema ${mapping.schema} locally`
```

Comment:
- avoid mentioning duplicate values in log messages, it makes them noisy
- consider prefix the message with a topic (as done in other parts of this file), it gives context

```suggestion
                    `YamlSchemaHandler: failed to assign schema to ${mapping.uri}. Failed to find schema locally: ${mapping.schema}`
```